kore/src/Kore/ASTVerifier/PatternVerifier.hs

```diff
@@ -708,9 +710,25 @@ verifyDomainValue domain = do
     let DomainValue { domainValueSort = patternSort } = domain
     Context { builtinDomainValueVerifiers, indexedModule } <- Reader.ask
     verifyPatternSort patternSort
-    let lookupSortDeclaration' sortId = do
+    let
+        lookupSortDeclaration' sortId = do
             (_, sortDecl) <- resolveSort indexedModule sortId
             return sortDecl
+        dvSortId = case patternSort of
+            SortVariableSort _ ->
+                error "Unimplemented: domain values with variable sorts"
+            SortActualSort SortActual {sortActualName} -> sortActualName
+    (sortAttrs, _) <- resolveSort indexedModule dvSortId
+    koreFailWithLocationsWhen
```

Comment:
Could we extract this to another function, say `verifySortHasDomainValues`, so that `verifyDomainValue` reads a bit better?